src/client/routes/Portfolio/index.tsx

```diff
@@ -24,124 +28,154 @@ const HeaderCard = styled(SummaryCard)`
   grid-column: 1 / 3;
 `;
 
-const Row = styled.div`
-  display: flex;
-  flex-wrap: wrap;
-  justify-content: flex-start;
-  grid-gap: ${({ theme }) => theme.layoutPadding};
-  flex-wrap: wrap;
+const StyledSpinnerLoading = styled(SpinnerLoading)`
   grid-column: 1 / 3;
+  flex: 1;
+  margin: 10rem 0;
 `;
 
-const StyledNoWalletCard = styled(NoWalletCard)`
-  grid-column: 1 / 3;
-  ${halfWidthCss}
+const RoleOption = styled.div<{ active?: boolean }>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 20rem;
+  height: 8rem;
+  border: 2px solid transparent;
+  color: ${({ theme }) => theme.colors.titles};
+  background-color: ${({ theme }) => theme.colors.background};
+  border-radius: ${({ theme }) => theme.globalRadius};
+  font-weight: 700;
+  cursor: pointer;
+
+  ${({ active, theme }) =>
+    active &&
+    `
+    background-color: ${theme.colors.backgroundVariant};
+    color: ${theme.colors.titlesVariant};
+    border-color: ${theme.colors.primary};
+  `}
 `;
 
-const StyledSummaryCard = styled(SummaryCard)`
-  width: 100%;
-  grid-column: 1 / 3;
-  ${halfWidthCss};
+const StyledSliderCard = styled(SliderCard)`
+  padding: 3rem;
+  margin: 0;
 `;
 
-const StyledSpinnerLoading = styled(SpinnerLoading)`
+const StyledBorrowerContainer = styled.div`
   grid-column: 1 / 3;
-  flex: 1;
-  margin: 10rem 0;
 `;
 
 export const Portfolio = () => {
   const { t } = useAppTranslation(['common', 'home']);
-  const { NETWORK_SETTINGS } = getConfig();
   const isMounting = useIsMounting();
+  const location = useLocation();
+  const dispatch = useAppDispatch();
+  const history = useHistory();
   const walletIsConnected = useAppSelector(WalletSelectors.selectWalletIsConnected);
 
-  const currentNetwork = useAppSelector(NetworkSelectors.selectCurrentNetwork);
-  const currentNetworkSettings = NETWORK_SETTINGS[currentNetwork];
-  const vaultsSummary = useAppSelector(VaultsSelectors.selectSummaryData);
   // const labsSummary = useAppSelector(LabsSelectors.selectSummaryData);
-  const walletSummary = useAppSelector(TokensSelectors.selectSummaryData);
   const userTokens = useAppSelector(TokensSelectors.selectUserTokens);
   const activeModal = useAppSelector(ModalSelectors.selectActiveModal);
   const appStatus = useAppSelector(AppSelectors.selectAppStatus);
   const tokensListStatus = useAppSelector(TokensSelectors.selectWalletTokensStatus);
   const generalLoading = (appStatus.loading || tokensListStatus.loading || isMounting) && !activeModal;
-  const userTokensLoading = generalLoading && !userTokens.length;
+  const userAddress: string | undefined = location.pathname.split('/')[2];
```

Comment:
should be able to get from react router history which will be much more reliable and reusable

Reply:
I typed and replaced with useParams()